### DISCUSSION TITLE
Windows arm64

### DISCUSSION THREAD

**Discussion by pb5050** [valinet/ExplorerPatcher#138] [General]
anyway you can do that? release it for arm64 anbd if anything i can do let me know

**Comment by valinet** [comment]
Moved to discussions as it is not an issue.

I have been asked previously, I do not own an arm64 machine, so I cannot test it. Some stuff has to be coded specIfically for that architecture. Furthermore, I also do not have too much time free time. Considering all of the above, the chances of me picking this up are slim.


Anyway, if you want to take a chance and attempt a port, I can help you with any questions regarding the code. Good luck!

**Comment by Amrsatrio** [comment]
![18F9D139-D9D8-4647-BFF7-A8B126D41E21](https://github.com/valinet/ExplorerPatcher/assets/20662640/a11dcc9e-23a5-439c-9f6f-a58c77097dac)

**Reply by Amrsatrio** [reply]
![image](https://github.com/valinet/ExplorerPatcher/assets/20662640/73682056-9fa5-45d7-b52c-af174d170705)

So far Win+X, Windows 10 Alt+Tab, and the reimplemented taskbar for 24H2 work. I will discuss first with @valinet before I can proceed further into release.

SWS crashes Explorer once enabled. ep_dwm (disable rounded corners) does nothing. Weather cannot be tested due to the stock 10 taskbar being no longer available and that I haven't implemented the people button in EP's taskbar.

**Reply by Amrsatrio** [reply]
![image](https://github.com/user-attachments/assets/ac25077c-9220-4517-ad44-c559d9cb14d5)

Worked on ep_dwm's rounded corners disabler.

**Reply by valinet** [reply]
@Amrsatrio I have just added you as a collaborator on the `ep_dwm` repo so you can contribute the ARM64 fixes there.

`sws` has to be looked into; maybe it is the language switching hooks it installs in order to respond properly for `Alt`+tilde shortcut - not core functionality but whatever (this also probably causes the language related bugs people report connected to `sws`). On one hand, maybe `sws` would live much better as a separate product, yet on the other hand ExplorerPatcher by its nature is a collection of "stuff" bundled together, so there's also that argument. And also the argument that **everything** would live way better as individual Windhawk mods...

**Reply by valinet** [reply]
@Amrsatrio By the way, my congratulations for the reimplemented taskbar, works and feels superb! Hard work pays off, great effort, my whole appreciation for it!

**Reply by valinet** [reply]
@Amrsatrio The in-house weather could work if instead of "abusing" the People button it would just instantiate its own button, but at the time I coded that, I did not know how to do so and it has since remained like this... Anyway, any non-essential feature (weather, sws, ep_dwm) could be marked as non-supported for ARM64 and should not get in the way of producing an actually usable ARM64 version.

**Reply by Amrsatrio** [reply]
Fixed SWS to the point it doesn't crash anymore. Now it runs into an infinite loop in _sws_WindowSwitcher_Calculate, which means it needs to be worked on to be able to be used under 24H2 even in the x64 version.

When you have time, can we have some private chats in our Discord? There are several stuff that I need your help on, and also several stuff that I'd like to say.
